### PR TITLE
fox_fo Decompiled & Data imported

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/overlays/ovl_i4/fox_fo.c [hunk 1]
```diff
@@ -1,29 +1,1513 @@
...
+                D_801784C8 = 58.0f;
+                D_801784FC = D_801784C8;
+                D_801784C8 = D_801784FC;
+                D_801784C8 = D_801784C8;
+                D_801784D4 = D_801784C8;
+                D_801784CC = 13.0f;
+                D_80178500 = D_801784CC;
+                D_801784CC = D_80178500;
+                D_801784CC = D_801784CC;
+                D_801784D8 = D_801784CC;
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:819]:
```suggestion
                D_801784D0 = D_801784C4 = D_801784C4 = D_801784F8 = D_801784C4 = -59.0f;
                D_801784D4 = D_801784C8 = D_801784C8 = D_801784FC = D_801784C8 = 58.0f;
                D_801784D8 = D_801784CC = D_801784CC = D_80178500 = D_801784CC = 13.0f;
```

I think these are likely chain assigns. Should reduce the padding needed for the stack too

File: src/overlays/ovl_i4/fox_fo.c [hunk 2]
```diff
@@ -1,29 +1,1513 @@
...
+                actorPtr[1].obj.rot.z = 0.0f;
+                actorPtr[1].obj.pos.x = 100.0f;
+                actorPtr[1].obj.pos.y = 40.0f;
+                actorPtr[2].obj.pos.y = 90.0f;
+                player->unk_1D0 = 22;
+            }
+
+            switch (gCsFrameCount) {
+                case 0x8C:
+                    func_8001D444(0U, 0x26U, 0U, 0xFFU);
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:1208]:
```suggestion
                    func_8001D444(0, 0x26, 0, 0xFF);
```
I think you can get rid of most of the Us you have here

File: src/overlays/ovl_i4/fox_fo.c [hunk 3]
```diff
@@ -1,29 +1,1513 @@
...
+                    break;
+                case 0x1C2:
+                    D_80177840 = 0x64;
+                    break;
+            }
+            src.x = 0.0f;
+            src.y = D_80177A48[7];
+            src.z = D_80177A48[5];
+            Matrix_Translate(gCalcMatrix, player->pos.x, 0.0f, player->pos.z + D_80177D20, (u8) 0);
+            Matrix_RotateY(gCalcMatrix, -(D_80177A48[4] * 0.017453292f), 1U);
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:1218]:
```suggestion
            Matrix_RotateY(gCalcMatrix, -(D_80177A48[4] * M_DTOR), 1);
```

Comment (inspectredc) [reply src/overlays/ovl_i4/fox_fo.c:1218]: similarly needs replacing in other areas

File: src/overlays/ovl_i4/fox_fo.c [hunk 4]
```diff
@@ -1,29 +1,1513 @@
...
+                D_80177840 = 0x64;
+            }
+            break;
+
+        case 11:
+            D_80177A48[1] += D_80177A48[3];
+            Matrix_RotateX(gCalcMatrix, -0.17453292f, 0);
+            Matrix_RotateY(gCalcMatrix, D_80177A48[1] * 0.017453292f, 1);
+            src.y = 0.0f;
+            src.x = 0;
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:905]: I think when i messed around with this earlier, you could instead have
```suggestion
            src.x = 0.0f;
            src.y = 0;
```
which seems somewhat more real

File: src/overlays/ovl_i4/fox_fo.c [hunk 5]
```diff
@@ -1,29 +1,1513 @@
...
+    Audio_PlaySfx(0x11030010U, actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+    Audio_PlaySfx(0x31024059U, actor->sfxPos, 0U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+}
+
+void func_i4_8018927C(Player* player) {
+    s32 i;
+    Vec3f src;
+    Vec3f dest;
+    Actor* actorPtr = &gActors[0];
+    Actor* actor2 = &gActors[2];
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:643]: I think actor2 is unused here. I'm also not sure what the general approach we should have for using the actorPtr like this is. I've been going with
```
Actor* actor0 = &gActors[0];
Actor* actor1 = &gActors[1];
Actor* actor2 = &gActors[2];
Actor* actor3 = &gActors[3];
```
and using these variables in place of actorPtr[x]. everywhere. It makes more sense to be written like this in my head so they could refer to each character but they both should match so idk

File: src/overlays/ovl_i4/fox_fo.c [hunk 6]
```diff
@@ -1,29 +1,1513 @@
...
+        actor->timer_0CA[0] = 0;
+        actor->info.bonus = 0;
+        Audio_PlaySfx(0x2903B009U, actor->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+    }
+}
+
+Vec3f D_i4_8019EE40 = { 0.0f, 0.0f, 0.0f };
+
+void func_i4_80188DA0(s32 arg0, Vec3f* arg1, void* ptr) {
+    Vec3f vec = D_i4_8019EE40;
```
Comment (inspectredc) [inline src/overlays/ovl_i4/fox_fo.c:514]:
```suggestion
void func_i4_80188DA0(s32 arg0, Vec3f* arg1, void* ptr) {
    Vec3f vec = { 0.0f, 0.0f, 0.0f };
```

**Review by inspectredc — COMMENTED** [review]
Looks good! Mostly just a few cleanup things that I think make sense for that final function

File: src/overlays/ovl_i4/fox_fo.c (hunk 5 above)
Comment (sonicdcer) [reply src/overlays/ovl_i4/fox_fo.c:643]: Yeah, i've been trying to do the same, although it doesn't seem to match in some parts, i'll see what i can do.

**Review by inspectredc — APPROVED** [review]

File: src/overlays/ovl_i4/fox_fo.c [hunk 7]
```diff
@@ -1,29 +1,1503 @@
...
+    if (D_8015F928 == 8540) {
+        Radio_PlayMessage(gMsg_ID_9400, RCID_ROB64);
+        func_8001D444(0, 0x800A, 0, 0);
+        gActors[7].unk_0E6 = -1;
+        gActors[6].unk_0E6 = gActors[7].unk_0E6;
+        gActors[5].unk_0E6 = gActors[7].unk_0E6;
+        gActors[4].unk_0E6 = gActors[7].unk_0E6;
+        gActors[3].unk_0E6 = gActors[7].unk_0E6;
+        gActors[2].unk_0E6 = gActors[7].unk_0E6;
+        gActors[1].unk_0E6 = gActors[7].unk_0E6;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_fo.c:163]: This must have been a chain assign. Should clear up the pads on the stack, too.

File: src/overlays/ovl_i4/fox_fo.c [hunk 8]
```diff
@@ -1,29 +1,1503 @@
...
+    actor->obj.pos.y = D_i4_8019EE5C[arg1] + gPlayer[0].pos.y;
+    actor->obj.pos.z = D_i4_8019EE6C[arg1] + gPlayer[0].unk_138;
+    actor->vel.z = gPlayer[0].unk_0D0;
+
+    Object_SetInfo(&actor->info, actor->obj.id);
+
+    if (arg1 < 3) {
+        actor->iwork[11] = 1;
+        Audio_PlaySfx(0x3100000CU, actor->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        return;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_fo.c:625]: else?

File: src/overlays/ovl_i4/fox_fo.c [hunk 9]
```diff
@@ -1,29 +1,1503 @@
...
+        case 6:
+            arg3->x += actor->fwork[0];
+            break;
+
+        case 7:
+            arg3->x += actor->fwork[0] * 0.7f;
+            break;
+    }
+
+    return 0;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_fo.c:578]:
```suggestion
    return false;
```

**Review by petrie911 — COMMENTED** [review]
just some small things

**Pull request merged**